src/Acuminator/Acuminator.Analyzers/FixProviders/DAC/ConstructorInGraphExtensionCodeFix.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Acuminator.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Acuminator.Analyzers.FixProviders
+{
+	[Shared]
+	[ExportCodeFixProvider(LanguageNames.CSharp)]
+	public class ConstructorInGraphExtensionCodeFix : CodeFixProvider
+	{
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(Descriptors.PX1040_ConstructorInGraphExtension.Id);
+
+		public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			if (context.CancellationToken.IsCancellationRequested) return;
+
+			var root = await context.Document.GetSyntaxRootAsync().ConfigureAwait(false);
+			var node = root.FindNode(context.Span)?.FirstAncestorOrSelf<ConstructorDeclarationSyntax>();
+
+			if (node != null)
+			{
+				string title = nameof(Resources.PX1040Fix).GetLocalized().ToString();
+				context.RegisterCodeFix(CodeAction.Create(title, 
+					c => MoveCodeFromConstructorToInitialize(context.Document, node, c), title),
+					context.Diagnostics);
+			}
+		}
+
+		private async Task<Document> MoveCodeFromConstructorToInitialize(
+			Document document,
+			ConstructorDeclarationSyntax constructorNode,
+			CancellationToken cancellationToken)
+		{
+			if (cancellationToken.IsCancellationRequested) return document;
+
+			var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+			var methodSymbol = semanticModel.GetDeclaredSymbol(constructorNode);
+			var newRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+			var initializeSymbol = methodSymbol.ContainingType
+				.GetMembers("Initialize")
+				.OfType<IMethodSymbol>()
+				.FirstOrDefault(m => m.IsOverride
+				                     && m.DeclaredAccessibility == Accessibility.Public
+				                     && m.ReturnsVoid
+				                     && m.Parameters.IsEmpty);
+
+			if (initializeSymbol != null)
+			{
+				// Copy the body from the constructor to the beggining of the existing Initialize() method
+				var initializeNode = await initializeSymbol
+					.GetSyntaxAsync(cancellationToken).ConfigureAwait(false) as MethodDeclarationSyntax;
+				if (initializeNode == null) return document;
+
+				var statements = constructorNode.Body.Statements.AddRange(initializeNode.Body.Statements);
```

Comment:
@dropsonic should the constructor statements be added to the end of Initialize? 
Is the Initialize method called inside base constructor?  If you consider the order of execution what will be called first - the code in the removed constructor or the code in the Initialize method? If the Initialize method is second then the fix is ok, otherwise the constructors statements I think should be inserted  after Initialize statements.
If the Initilize method is called in a base constructor of BLC extension then the actual order should be like this:
base constructors -> Initialize() in one of them -> constructor. 

Also there could be name conflicts if both methods declare local variables with same names.

Reply:
`Initialize` method is second. This order was made on purpose.
We don't have a goal to make it work in each case; if there are conflicts, the user should resolve them after applying the code fix.